src/bin/miri.rs

```diff
@@ -170,7 +176,15 @@ fn main() {
         rustc_driver::init_rustc_env_logger();
         // We cannot use `rustc_driver::main` as we need to adjust the CLI arguments.
         let mut callbacks = rustc_driver::TimePassesCallbacks::default();
-        run_compiler(env::args().collect(), &mut callbacks)
+        run_compiler(
+            env::args().collect(),
+            &mut callbacks,
+            // Don't insert `MIRI_DEFAULT_ARGS`, in particular, `--cfg=miri`, if we are building a
+            // "host" crate. That may cause procedural macros (and probably build scripts) to depend
+            // on Miri-only symbols, such as `miri_resolve_frame`:
+            // https://github.com/rust-lang/miri/issues/1760
+            env::args().skip(1).any(|arg| arg == "--target"),
```

Comment:
Doesn't `cargo miri` already know if this is a crate built for interpretation vs for the host? Would it be easier to forward that information instead of re-computing it by checking for a command-line flag?

Reply:
I imagine something like `MIRI_BE_RUSTC=host` disabling the arguments.